Source/RapyutaSimulationPlugins/Private/Drives/DifferentialDriveComponent.cpp

```diff
@@ -39,59 +41,107 @@ void UDifferentialDriveComponent::UpdateMovement(float DeltaTime)
         float velL = Velocity.X + AngularVelocity.Z * WheelSeparationHalf;
         float velR = Velocity.X - AngularVelocity.Z * WheelSeparationHalf;
 
-
         WheelLeft->SetAngularVelocityTarget(FVector(-velL / WheelPerimeter, 0, 0));
         WheelRight->SetAngularVelocityTarget(FVector(-velR / WheelPerimeter, 0, 0));
         WheelLeft->SetAngularDriveParams(MaxForce, MaxForce, MaxForce);
         WheelRight->SetAngularDriveParams(MaxForce, MaxForce, MaxForce);
-
-        //  UE_LOG(LogDifferentialDriveComponent, Error, TEXT("%f, %f, %f, %f"), AngularVelocity.Z,);
-    }

```

Comment:
I actually did not want to remove but somehow probably with the new dynamically created Move component, in case of Physics Waffle, BP variables as the passed-in Wheels here get garbarged and so those warnings get printed along with the valid ones. May you try to see also?

Reply:
I see two differentialdrivecomponents are spawned and one is properly set but another is not.
please check how to avoid thata

Reply:
* Yeah, because of creating components under `OnConstruction`, so as running in the Editor, it would cause the object to be repeatedly created and only the latest one is used while previous ones get garbaged. So in that case, an invalid `WheelLeft/Right` just means their owner actor get garbaged and might not necessarily mean an error.
* We could not create MoveComponent in BeginPlay() since it would be late for its initialization, which must happen before `AIController::OnPossess()` for its ROS publisher to be initialized,
So, I haven't found another better overridable method of [these](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/Engine/Classes/GameFramework/Actor.h#L100) for that purpose..

Reply:
@chalice-graeme do you have any idea to avoid repeatdly creation of objects?